W-20897660: Forward compatibility (#298)
mulesoft/docs-mule-sdk: modules/ROOT/pages/forward-compatibility-implicit-params.adoc

= Implicit Parameters
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: mule, sdk, forward compatibility, implicit parameters

> Implicit parameters are parameters that the Mule runtime engine automatically injects without requiring configuration from the application developer. Examples include `StreamingHelper`, `ComponentLocation`, and `CorrelationInfo`. These parameters do not appear in the Extension Model.

Reviewer: ```suggestion
Implicit parameters are parameters that the Mule runtime engine automatically injects without requiring configuration from the application developer. Examples include `StreamingHelper`, `ComponentLocation`, and `CorrelationInfo`. These parameters don't appear in the Extension Model.
```

== Implicit Parameter Types from org.mule.sdk.api

Starting with Mule 4.5, the runtime automatically treats parameters with types from `org.mule.sdk.api` and its subpackages as implicit, enabling forward compatibility for new implicit parameter types.

When you use a new implicit parameter type that was introduced in a later Mule version:

* On runtimes that support the type, the parameter is injected normally

> * On earlier Mule runtime versions that don't recognize the type, the parameter resolves to `null` and the runtime treats the parameter as implicit (it doesn't appear in the Extension Model and can't be configured by the application developer).

Reviewer: ```suggestion
* On earlier Mule runtime versions that don't recognize the type, the parameter resolves to `null` and the runtime treats the parameter as implicit. Mule runtime treating a parameter as implicit means that it doesn't appear in the Extension Model and can't be configured by the application developer.
```
